File: audit_game.js
const fs = require('fs');
const content = fs.readFileSync('C:/Users/Administrator/Documents/kimi/workspace/farm_game.html', 'utf8');
const scriptMatch = content.match(/<script>([\s\S]*?)<\/script>/);
const code = scriptMatch[1];

const vm = require('vm');
const sandbox = {
  console, Math, JSON, Date, Array, Object, String, Number, RegExp, Error,
  window: {}, document: {}, localStorage: { getItem: () => null, setItem: () => {}, removeItem: () => {} },
  setTimeout: () => {}, clearTimeout: () => {}, setInterval: () => {}, clearInterval: () => {}
};
vm.createContext(sandbox);
vm.runInContext(code, sandbox, { timeout: 5000 });

const audit = { issues: [], checks: [], data: {} };

function push(msg) { audit.checks.push(msg); }

// ============ CROP_DATA ============
push('=== CROP_DATA ===');
const crops = sandbox.CROP_DATA;
for (const [k, c] of Object.entries(crops)) {
  const price = c.type === 'rice' ? 3.0 : 1.5;
  const net = c.baseYield * price - c.seedPrice;
  const daily = net / c.growDays;
  push(`${c.name}: cost=${c.seedPrice}, yield=${c.baseYield}, price=${price}, net=${net}, daily=${daily.toFixed(1)}`);
  audit.data[k] = { net, daily, growDays: c.growDays };
  if (c.steps.length !== c.stepCosts.length) {
    audit.issues.push(`ERROR: ${c.name} steps(${c.steps.length}) != stepCosts(${c.stepCosts.length})`);
  }
  const stepTotal = c.stepCosts.reduce((a, b) => a + b, 0);
  push(`  steps=${c.steps.length}, stepTotal=${stepTotal}, avg=${(stepTotal/c.stepCosts.length).toFixed(1)}`);
}

// ============ HOUSE_DATA ============
push('');
push('=== HOUSE_DATA ===');
const houses = sandbox.HOUSE_DATA;
for (const [k, h] of Object.entries(houses)) {
  const dailyRegen = (h.staminaRegen / 4) * 24;
  push(`${h.name}: staminaRegen=${h.staminaRegen}/hr => daily=${dailyRegen}, healthRegen=${h.healthRegen}/night, rent=${h.rent}, upgrade=${h.upgradeCost}`);
  if (dailyRegen >= 120) audit.issues.push(`WARNING: ${h.name} daily regen ${dailyRegen} >= max stamina 120`);
}

// ============ TOOL_DATA ============
push('');
push('=== TOOL_DATA ===');
const tools = sandbox.TOOL_DATA;
for (const [k, t] of Object.entries(tools)) {
  push(`${t.name}: ${t.price}元, effect='${t.effect}', type=${t.type}`);
}

// ============ WEATHER_DATA ============
push('');
push('=== WEATHER_DATA ===');
const weather = sandbox.WEATHER_DATA;
const seasonWeather = sandbox.SEASON_WEATHER;
for (const [k, w] of Object.entries(weather)) {
  push(`${w.name}: ${w.emoji}, stamina=${w.stamina}, growth=${w.growth}, duration=${JSON.stringify(w.duration)}`);
}
for (const [s, w] of Object.entries(seasonWeather)) {
  push(`Season ${s}: ${w.join(', ')}`);
}

// ============ EVENTS ============
push('');
push('=== EVENTS ===');
const events = sandbox.EVENTS || [];
const storyEvents = sandbox.STORY_EVENTS || [];
push(`Events count: ${events.length}`);
push(`StoryEvents count: ${storyEvents.length}`);
if (events.length > 0) {
  const probs = events.map(e => e.probability || 0);
  const totalProb = probs.reduce((a, b) => a + b, 0);
  push(`Total event probability: ${totalProb.toFixed(3)}`);
  if (totalProb > 1.0) audit.issues.push(`WARNING: Total event probability ${totalProb} > 1.0`);
  const disasters = events.filter(e => e.type === 'disaster').length;
  const gains = events.filter(e => e.type === 'gain').length;
  push(`Disaster events: ${disasters}, Gain events: ${gains}`);
  if (disasters > gains * 1.5) audit.issues.push(`WARNING: Too many disasters (${disasters}) vs gains (${gains})`);
}

// ============ NPC_DATA ============
push('');
push('=== NPC_DATA ===');
const npcs = sandbox.NPC_DATA;
for (const [k, n] of Object.entries(npcs)) {
  push(`${n.name}: ${n.emoji}, gift=${n.gift}, giftFreq=${n.giftFrequency}, initialFavor=${n.initialFavor}`);
}

// ============ TECH_DATA ============
push('');
push('=== TECH_DATA ===');
const techs = sandbox.TECH_DATA;
let techCount = 0;
for (const [cat, list] of Object.entries(techs)) {
  techCount += list.length;
  push(`Category ${cat}: ${list.length} techs`);
  for (const t of list) {
    push(`  ${t.name}: cost=${t.cost}, ${t.description}`);
  }
}

// ============ FORAGING_DATA ============
push('');
push('=== FORAGING_DATA ===');
const foraging = sandbox.FORAGING_DATA || {};
for (const [k, f] of Object.entries(foraging)) {
  push(`${k}: ${f.items.length} items, staminaCost=${f.staminaCost}, avgPrice=${f.items.reduce((a,i)=>a+(i.price||0),0)/f.items.length}`);
}

// ============ PROCESSING_DATA ============
push('');
push('=== PROCESSING_DATA ===');
const processing = sandbox.PROCESSING_DATA || {};
for (const [k, p] of Object.entries(processing)) {
  push(`${k}: input=${p.inputAmount} ${p.inputCrop}, cost=${p.processingCost}, output=${p.outputAmount} ${p.outputItem}`);
}

// ============ DAILY_ACTIVITIES ============
push('');
push('=== DAILY_ACTIVITIES ===');
const activities = sandbox.DAILY_ACTIVITIES || {};
for (const [k, a] of Object.entries(activities)) {
  push(`${a.name}: stamina-${a.staminaCost}, ${a.description}`);
}

// ============ PET_DATA ============
push('');
push('=== PET_DATA ===');
const pets = sandbox.PET_DATA || {};
for (const [k, p] of Object.entries(pets)) {
  push(`${p.name}: cost=${p.cost}, food=${p.dailyFoodCost}, unlock=${p.unlockCondition}`);
}

// ============ BUILDING_DATA ============
push('');
push('=== BUILDING_DATA ===');
const buildings = sandbox.BUILDING_DATA || {};
for (const [k, b] of Object.entries(buildings)) {
  push(`${b.name}: buildCost=${b.buildCost}, dailyIncome=${b.dailyIncome}, unlock=${b.unlockCondition}`);
}

// ============ TAB_UNLOCK_CONFIG ============
push('');
push('=== TAB_UNLOCK_CONFIG ===');
const tabs = sandbox.TAB_UNLOCK_CONFIG || {};
for (const [k, t] of Object.entries(tabs)) {
  push(`${t.name}: day=${t.day}${t.money ? ', money=' + t.money : ''}`);
}

// ============ SKILL_EFFECTS ============
push('');
push('=== SKILL_EFFECTS ===');
const skillEffects = sandbox.SKILL_EFFECTS || {};
for (const [k, s] of Object.entries(skillEffects)) {
  push(`${s.name}: ${s.description}`);
}

// ============ NPC_MILESTONES ============
push('');
push('=== NPC_MILESTONES ===');
const milestones = sandbox.NPC_MILESTONES || {};
for (const [k, m] of Object.entries(milestones)) {
  push(`${k}: ${m.length} milestones`);
}

// ============ QUEST_DATA ============
push('');
push('=== QUEST_DATA ===');
const quests = sandbox.QUEST_DATA || {};
const questCount = Object.keys(quests).length;
push(`Quests: ${questCount}`);
for (const [k, q] of Object.entries(quests)) {
  push(`${q.name}: ${q.description}, reward=${JSON.stringify(q.reward)}`);
}

// ============ MILESTONE_DATA ============
push('');
push('=== MILESTONE_DATA ===');
const mData = sandbox.MILESTONE_DATA || {};
for (const [k, m] of Object.entries(mData)) {
  push(`${m.name}: ${m.description}, reward=${JSON.stringify(m.reward)}`);
}

// ============ CROSS-CHECKS ============
push('');
push('=== CROSS-CHECKS ===');

// House upgrade chain
const houseOrder = ['新手房', '进阶房', '精英房', '终极房'];
for (let i = 1; i < houseOrder.length; i++) {
  const prev = houses[houseOrder[i-1]];
  const curr = houses[houseOrder[i]];
  if (!curr) audit.issues.push(`ERROR: Missing house ${houseOrder[i]}`);
  if (curr && curr.upgradeCost <= prev.upgradeCost) {
    audit.issues.push(`WARNING: ${curr.name} upgradeCost(${curr.upgradeCost}) <= ${prev.name}(${prev.upgradeCost})`);
  }
  if (curr && curr.rent <= prev.rent) {
    audit.issues.push(`WARNING: ${curr.name} rent(${curr.rent}) <= ${prev.name}(${prev.rent})`);
  }
}

// Check season consistency
for (const [k, c] of Object.entries(crops)) {
  if (!['spring', 'summer', 'autumn', 'winter'].includes(c.season)) {
    audit.issues.push(`ERROR: ${c.name} has invalid season '${c.season}'`);
  }
}

// Check tool effect descriptions for consistency
for (const [k, t] of Object.entries(tools)) {
  if (t.effect.includes('-自动') || t.effect.includes('自动')) {
    audit.issues.push(`WARNING: Tool ${t.name} has 'automatic' effect: '${t.effect}'`);
  }
}

// Check foraging item prices vs stamina cost
for (const [k, f] of Object.entries(foraging)) {
  const avgPrice = f.items.reduce((a, i) => a + (i.price || 0), 0) / f.items.length;
  if (avgPrice < f.staminaCost * 2) {
    audit.issues.push(`WARNING: Foraging ${k} avgPrice(${avgPrice}) < staminaCost*2(${f.staminaCost * 2}), may not be worth it`);
  }
}

// Check building ROI
for (const [k, b] of Object.entries(buildings)) {
  if (b.dailyIncome > 0 && b.buildCost > 0) {
    const daysToROI = b.buildCost / b.dailyIncome;
    push(`${b.name}: ROI in ${daysToROI.toFixed(1)} days`);
    if (daysToROI > 60) audit.issues.push(`WARNING: ${b.name} ROI too long: ${daysToROI.toFixed(1)} days`);
    if (daysToROI < 3) audit.issues.push(`WARNING: ${b.name} ROI too short: ${daysToROI.toFixed(1)} days`);
  }
}

// Check pet costs vs starting money
const startMoney = 1500;
for (const [k, p] of Object.entries(pets)) {
  if (p.cost > startMoney * 2) {
    audit.issues.push(`WARNING: Pet ${p.name} cost(${p.cost}) > 2x starting money(${startMoney})`);
  }
  const dailyNet = p.dailyFoodCost; // Assume no other income from pet
  if (dailyNet > 50) audit.issues.push(`WARNING: Pet ${p.name} daily cost(${dailyNet}) too high`);
}

// Check tech costs
let allTechs = [];
for (const [cat, list] of Object.entries(techs)) {
  allTechs = allTechs.concat(list);
}
const totalTechCost = allTechs.reduce((a, t) => a + t.cost, 0);
push(`Total tech cost: ${totalTechCost}`);
const avgTechCost = totalTechCost / allTechs.length;
push(`Average tech cost: ${avgTechCost}`);
if (totalTechCost > 100000) audit.issues.push(`WARNING: Total tech cost ${totalTechCost} too high`);

// Summary
push('');
push('=== SUMMARY ===');
push(`Total issues found: ${audit.issues.length}`);
if (audit.issues.length > 0) {
  push('ISSUES:');
  audit.issues.forEach(i => push('  ' + i));
}

console.log(audit.checks.join('\n'));

File: audit_game.py
import json, re

with open('C:/Users/Administrator/Documents/kimi/workspace/farm_game.html', 'r', encoding='utf-8') as f:
    content = f.read()

script_match = re.search(r'<script>(.*?)</script>', content, re.DOTALL)
code = script_match.group(1)

issues = []
checks = []

def extract_object(name):
    pattern = r'const\s+' + re.escape(name) + r'\s*=\s*\{[\s\S]*?\};\s*(?=\nconst|\nfunction|\n//|\n\n)'
    match = re.search(pattern, code)
    if match: return match.group(0)
    pattern2 = r'const\s+' + re.escape(name) + r'\s*=\s*\[[\s\S]*?\];'
    match2 = re.search(pattern2, code)
    if match2: return match2.group(0)
    return None

def js_to_dict(js_code):
    try:
        py_code = re.sub(r'const\s+\w+\s*=\s*', '', js_code)
        py_code = re.sub(r'\btrue\b', 'True', py_code)
        py_code = re.sub(r'\bfalse\b', 'False', py_code)
        py_code = re.sub(r'\bnull\b', 'None', py_code)
        py_code = re.sub(r'\bundefined\b', 'None', py_code)
        py_code = py_code.rstrip().rstrip(';')
        return eval(py_code)
    except:
        return None

def js_to_list(js_code):
    try:
        py_code = re.sub(r'const\s+\w+\s*=\s*', '', js_code)
        py_code = re.sub(r'\btrue\b', 'True', py_code)
        py_code = re.sub(r'\bfalse\b', 'False', py_code)
        py_code = re.sub(r'\bnull\b', 'None', py_code)
        py_code = py_code.rstrip().rstrip(';')
        return eval(py_code)
    except:
        return None

checks.append('=== CROP_DATA ===')
crop_data = extract_object('CROP_DATA')
crops = js_to_dict(crop_data) if crop_data else {}
for k, c in crops.items():
    price = 3.0 if c['type'] == 'rice' else 1.5
    net = c['baseYield'] * price - c['seedPrice']
    daily = net / c['growDays']
    checks.append(f"{c['name']}: seed={c['seedPrice']}, yield={c['baseYield']}, price={price}, net={net}, daily={daily:.1f}")
    if len(c['steps']) != len(c['stepCosts']):
        issues.append(f"ERROR: {c['name']} steps({len(c['steps'])}) != stepCosts({len(c['stepCosts'])})")
    step_total = sum(c['stepCosts'])
    checks.append(f"  steps={len(c['steps'])}, stepTotal={step_total}, avg={step_total/len(c['stepCosts']):.1f}")
    if c['season'] not in ['spring','summer','autumn','winter']:
        issues.append(f"ERROR: {c['name']} invalid season: {c['season']}")

checks.append('')
checks.append('=== HOUSE_DATA ===')
house_data = extract_object('HOUSE_DATA')
houses = js_to_dict(house_data) if house_data else {}
for k, h in houses.items():
    daily_regen = (h['staminaRegen'] / 4) * 24
    checks.append(f"{h['name']}: staminaRegen={h['staminaRegen']}/hr => daily={daily_regen}, healthRegen={h['healthRegen']}/night, rent={h['rent']}, upgrade={h['upgradeCost']}")
    if daily_regen >= 120:
        issues.append(f"WARNING: {h['name']} daily regen {daily_regen} >= max stamina 120")

checks.append('')
checks.append('=== TOOL_DATA ===')
tool_data = extract_object('TOOL_DATA')
tools = js_to_dict(tool_data) if tool_data else {}
for k, t in tools.items():
    checks.append(f"{t['name']}: {t['price']}元, effect='{t['effect']}', type={t['type']}")
    if '自动' in t['effect']:
        issues.append(f"WARNING: Tool {t['name']} has 'auto' effect: {t['effect']}")

checks.append('')
checks.append('=== WEATHER_DATA ===')
weather_data = extract_object('WEATHER_DATA')
weather = js_to_dict(weather_data) if weather_data else {}
for k, w in weather.items():
    checks.append(f"{w['name']}: {w['emoji']}, stamina={w.get('stamina','?')}, growth={w.get('growth','?')}, duration={w.get('duration','?')}")

checks.append('')
checks.append('=== SEASON_WEATHER ===')
sw_data = extract_object('SEASON_WEATHER')
sw = js_to_dict(sw_data) if sw_data else {}
for s, w in sw.items():
    checks.append(f"Season {s}: {', '.join(w)}")

checks.append('')
checks.append('=== EVENTS ===')
events_data = extract_object('EVENTS')
events = js_to_dict(events_data) if events_data else {}
if events:
    checks.append(f"Events count: {len(events)}")
    probs = [e.get('probability', 0) for e in events.values()]
    total_prob = sum(probs)
    checks.append(f"Total event probability: {total_prob:.3f}")
    if total_prob > 1.0:
        issues.append(f"WARNING: Total event probability {total_prob} > 1.0")
    disasters = sum(1 for e in events.values() if e.get('type') == 'disaster')
    gains = sum(1 for e in events.values() if e.get('type') == 'gain')
    checks.append(f"Disasters: {disasters}, Gains: {gains}")
    if disasters > gains * 1.5:
        issues.append(f"WARNING: Too many disasters ({disasters}) vs gains ({gains})")

checks.append('')
checks.append('=== STORY_EVENTS ===')
story_data = extract_object('STORY_EVENTS')
story = js_to_list(story_data) if story_data else []
if story:
    checks.append(f"Story events count: {len(story)}")
    probs = [e.get('probability', 0) for e in story]
    total_prob = sum(probs)
    checks.append(f"Total story probability: {total_prob:.3f}")
    if total_prob > 1.0:
        issues.append(f"WARNING: Total story probability {total_prob} > 1.0")

checks.append('')
checks.append('=== NPC_DATA ===')
npc_data = extract_object('NPC_DATA')
npcs = js_to_dict(npc_data) if npc_data else {}
for k, n in npcs.items():
    checks.append(f"{n['name']}: {n['emoji']}, gift={n.get('gift','?')}, giftFreq={n.get('giftFrequency','?')}, initialFavor={n.get('initialFavor','?')}")

checks.append('')
checks.append('=== TECH_DATA ===')
tech_data = extract_object('TECH_DATA')
techs = js_to_dict(tech_data) if tech_data else {}
if techs:
    total_cost = 0
    total_count = 0
    for cat, list in techs.items():
        total_count += len(list)
        for t in list:
            total_cost += t.get('cost', 0)
            checks.append(f"  {t['name']}: cost={t.get('cost','?')}")
    checks.append(f"Total techs: {total_count}, total cost: {total_cost}, avg: {total_cost/total_count:.0f}")
    if total_cost > 100000:
        issues.append(f"WARNING: Total tech cost {total_cost} too high")

checks.append('')
checks.append('=== FORAGING_DATA ===')
foraging_data = extract_object('FORAGING_DATA')
foraging = js_to_dict(foraging_data) if foraging_data else {}
for k, f in foraging.items():
    avg_price = sum(i.get('price', 0) for i in f['items']) / len(f['items'])
    checks.append(f"{k}: {len(f['items'])} items, stamina={f['staminaCost']}, avgPrice={avg_price:.1f}")
    if avg_price < f['staminaCost'] * 2:
        issues.append(f"WARNING: Foraging {k} avgPrice({avg_price:.1f}) < staminaCost*2({f['staminaCost']*2})")

checks.append('')
checks.append('=== PROCESSING_DATA ===')
proc_data = extract_object('PROCESSING_DATA')
proc = js_to_dict(proc_data) if proc_data else {}
for k, p in proc.items():
    checks.append(f"{k}: input={p['inputAmount']} {p['inputCrop']}, cost={p['processingCost']}, output={p['outputAmount']} {p['outputItem']}")

checks.append('')
checks.append('=== DAILY_ACTIVITIES ===')
act_data = extract_object('DAILY_ACTIVITIES')
acts = js_to_dict(act_data) if act_data else {}
for k, a in acts.items():
    checks.append(f"{a['name']}: stamina-{a['staminaCost']}, {a['description']}")

checks.append('')
checks.append('=== PET_DATA ===')
pet_data = extract_object('PET_DATA')
pets = js_to_dict(pet_data) if pet_data else {}
for k, p in pets.items():
    checks.append(f"{p['name']}: cost={p['cost']}, food={p['dailyFoodCost']}, unlock={p['unlockCondition']}")
    if p['cost'] > 3000:
        issues.append(f"WARNING: Pet {p['name']} cost {p['cost']} too high vs start money 1500")

checks.append('')
checks.append('=== BUILDING_DATA ===')
build_data = extract_object('BUILDING_DATA')
builds = js_to_dict(build_data) if build_data else {}
for k, b in builds.items():
    income = b.get('dailyIncome', 0)
    cost = b.get('buildCost', 0)
    if income > 0 and cost > 0:
        roi = cost / income
        checks.append(f"{b['name']}: buildCost={cost}, dailyIncome={income}, ROI={roi:.1f} days")
        if roi > 60:
            issues.append(f"WARNING: {b['name']} ROI too long: {roi:.1f} days")
        if roi < 3:
            issues.append(f"WARNING: {b['name']} ROI too short: {roi:.1f} days")
    else:
        checks.append(f"{b['name']}: buildCost={cost}, dailyIncome={income}")

checks.append('')
checks.append('=== TAB_UNLOCK_CONFIG ===')
tab_data = extract_object('TAB_UNLOCK_CONFIG')
tabs = js_to_dict(tab_data) if tab_data else {}
for k, t in tabs.items():
    checks.append(f"{t['name']}: day={t.get('day','?')}, money={t.get('money', 'N/A')}")

checks.append('')
checks.append('=== SKILL_EFFECTS ===')
skill_data = extract_object('SKILL_EFFECTS')
skills = js_to_dict(skill_data) if skill_data else {}
for k, s in skills.items():
    checks.append(f"{s['name']}: {s['description']}")

checks.append('')
checks.append('=== QUEST_DATA ===')
quest_data = extract_object('QUEST_DATA')
quests = js_to_dict(quest_data) if quest_data else {}
if quests:
    checks.append(f"Quests: {len(quests)}")
    for k, q in quests.items():
        checks.append(f"  {q['name']}: reward={q.get('reward',{})}")

checks.append('')
checks.append('=== MILESTONE_DATA ===')
mile_data = extract_object('MILESTONE_DATA')
miles = js_to_dict(mile_data) if mile_data else {}
for k, m in miles.items():
    checks.append(f"{m['name']}: {m['description']}, reward={m.get('reward',{})}")

checks.append('')
checks.append('=== NPC_MILESTONES ===')
nm_data = extract_object('NPC_MILESTONES')
nms = js_to_dict(nm_data) if nm_data else {}
for k, m in nms.items():
    checks.append(f"{k}: {len(m)} milestones")

checks.append('')
checks.append('=== SKILL_EXP_TABLE ===')
exp_table = extract_object('SKILL_EXP_TABLE')
exp = js_to_list(exp_table) if exp_table else []
if exp:
    checks.append(f"Exp table: {exp}")
    for i in range(1, len(exp)):
        diff = exp[i] - exp[i-1]
        checks.append(f"  Lv{i}->{i+1}: {diff} exp")
    ratios = []
    for i in range(2, len(exp)):
        if exp[i-1] > 0:
            ratios.append(exp[i] / exp[i-1])
    if ratios:
        avg_ratio = sum(ratios) / len(ratios)
        checks.append(f"  Avg growth ratio: {avg_ratio:.2f}")
        if avg_ratio < 1.3:
            issues.append(f"WARNING: Skill exp growth too slow: avg ratio {avg_ratio:.2f}")

checks.append('')
checks.append('=== PET_FRIENDSHIP_LEVELS ===')
pet_levels = extract_object('PET_FRIENDSHIP_LEVELS')
levels = js_to_list(pet_levels) if pet_levels else []
if levels:
    checks.append(f"Pet levels: {levels}")

checks.append('')
checks.append('=== CROSS-CHECKS ===')
if houses:
    house_order = ['新手房', '进阶房', '精英房', '终极房']
    for i in range(1, len(house_order)):
        prev = houses.get(house_order[i-1])
        curr = houses.get(house_order[i])
        if curr and prev:
            if curr['upgradeCost'] <= prev['upgradeCost']:
                issues.append(f"WARNING: {curr['name']} upgradeCost({curr['upgradeCost']}) <= {prev['name']}({prev['upgradeCost']})")
            if curr['rent'] <= prev['rent']:
                issues.append(f"WARNING: {curr['name']} rent({curr['rent']}) <= {prev['name']}({prev['rent']})")

print('\n'.join(checks))
print(f"\n=== SUMMARY: {len(issues)} issues found ===")
for i in issues:
    print(f"  {i}")
